File: backend/app/schemas/company.py
# Company 관련 스키마
from datetime import datetime
from typing import Literal, Optional
from pydantic import BaseModel, Field, field_validator
import uuid


class CompanyBase(BaseModel):
    """Company 기본 스키마"""
    business_name: str = Field(..., min_length=1, max_length=200, description="사업장명")
    business_number: str = Field(..., pattern=r"^\d{3}-\d{2}-\d{5}$", description="사업자등록번호 (xxx-xx-xxxxx)")
    representative_name: str = Field(..., min_length=1, max_length=100, description="대표자명")
    industry_type: Literal["manufacturing", "food_service", "retail", "service", "it", "construction", "healthcare", "other"] = Field(..., description="업종")
    employee_count: int = Field(..., ge=0, le=1000, description="직원 수")
    address: Optional[str] = Field(None, description="주소")
    postal_code: Optional[str] = Field(None, description="우편번호")
    phone: Optional[str] = Field(None, description="대표 전화번호")

    @field_validator("business_number")
    @classmethod
    def validate_business_number(cls, v: str) -> str:
        """사업자등록번호 형식 검증 (공백 제거)"""
        v = v.strip()
        return v


class CompanyCreate(CompanyBase):
    """Company 생성"""
    pass


class CompanyUpdate(BaseModel):
    """Company 수정"""
    business_name: Optional[str] = Field(None, min_length=1, max_length=200)
    representative_name: Optional[str] = Field(None, min_length=1, max_length=100)
    industry_type: Optional[Literal["manufacturing", "food_service", "retail", "service", "it", "construction", "healthcare", "other"]] = None
    employee_count: Optional[int] = Field(None, ge=0, le=1000)
    address: Optional[str] = None
    postal_code: Optional[str] = None
    phone: Optional[str] = None


class CompanyDeleteRequest(BaseModel):
    """Company 삭제 요청"""
    confirmation: str = Field(..., description="사업장명 확인 (삭제 확인용)")


class CompanySelectRequest(BaseModel):
    """Company 선택 요청"""
    pass  # 빈 스키마


class CompanyResponse(CompanyBase):
    """Company 응답"""
    id: uuid.UUID
    owner_id: uuid.UUID
    work_rule_required: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class CompanyListItem(CompanyBase):
    """Company 목록 아이템"""
    id: uuid.UUID
    work_rule_required: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class PaginationMeta(BaseModel):
    """페이지네이션 메타"""
    cursor: Optional[str] = Field(None, description="다음 페이지 커서")
    hasNext: bool = Field(..., description="다음 페이지 존재 여부")
    limit: int = Field(..., ge=1, le=100, description="페이지 크기")
    totalCount: int = Field(..., ge=0, description="전체 개수")


class CompanyListResponse(BaseModel):
    """Company 목록 응답"""
    data: list[CompanyListItem]
    pagination: PaginationMeta


class TokenResponse(BaseModel):
    """토큰 응답 (사업장 선택 시)"""
    access_token: str
    refresh_token: str
    token_type: str = "bearer"
    expires_in: int
    company: CompanyListItem


class CompanySelectResponse(BaseModel):
    """사업장 선택 응답"""
    access_token: str
    refresh_token: str
    token_type: str = "bearer"
    expires_in: int
    company: CompanyListItem

File: frontend/types/company.ts
/**
 * 사업장 관련 타입 정의
 */

// 업종 타입
export type IndustryType =
  | 'manufacturing'   // 제조업
  | 'food_service'    // 요식업
  | 'retail'          // 소매업
  | 'service'         // 서비스업
  | 'it'              // IT/정보통신
  | 'construction'    // 건설업
  | 'healthcare'      // 의료업
  | 'other';          // 기타

// 사업장 정보
export interface Company {
  id: string;
  owner_id: string;
  business_name: string;
  business_number: string;
  representative_name: string;
  industry_type: IndustryType;
  employee_count: number;
  address?: string;
  postal_code?: string;
  phone?: string;
  work_rule_required: boolean;
  created_at: string;
  updated_at: string;
}

// 사업장 등록 요청
export interface CreateCompanyRequest {
  business_name: string;
  business_number: string;
  representative_name: string;
  industry_type: IndustryType;
  employee_count: number;
  address?: string;
  postal_code?: string;
  phone?: string;
}

// 사업장 수정 요청
export interface UpdateCompanyRequest {
  business_name: string;
  representative_name: string;
  industry_type: IndustryType;
  employee_count: number;
  address?: string;
  postal_code?: string;
  phone?: string;
}

// 사업장 삭제 요청
export interface DeleteCompanyRequest {
  confirmation: string;
}

// 사업장 목록 응답
export interface CompanyListResponse {
  companies: Company[];
  pagination: {
    cursor: string | null;
    hasNext: boolean;
    limit: number;
    totalCount: number;
  };
}

// 사업장 선택 응답
export interface SelectCompanyResponse {
  access_token: string;
  refresh_token: string;
  token_type: string;
  expires_in: number;
  company: {
    id: string;
    business_name: string;
  };
}

// 업종 옵션
export const INDUSTRY_OPTIONS: Array<{ value: IndustryType; label: string }> = [
  { value: 'manufacturing', label: '제조업' },
  { value: 'food_service', label: '요식업' },
  { value: 'retail', label: '소매업' },
  { value: 'service', label: '서비스업' },
  { value: 'it', label: 'IT/정보통신' },
  { value: 'construction', label: '건설업' },
  { value: 'healthcare', label: '의료업' },
  { value: 'other', label: '기타' },
];

// 업종 라벨 변환 함수
export function getIndustryLabel(type: IndustryType): string {
  const option = INDUSTRY_OPTIONS.find(opt => opt.value === type);
  return option?.label || type;
}
